doc/manual/chapters/mpcal.tex

```diff
@@ -0,0 +1,185 @@
+\end{itemize}
+
+Conversely, here is a list of similarities between archetypes and processes
+\begin{itemize}
+    \item Same labeling rules apply;
+    \item Archetypes have access to an implicit, immutable self parameter, defined when archetypes are instantiated.
+\end{itemize}
+
+\subsubsection{Instantiating for model checking}
+
+Since archetypes do not provide much of the information necessary for model checking themselves, we must be able to provide this information separately in order to generate a complete model checking scenario.
+
+We do this via a variation of the \lstinline|process| declaration called an \lstinline|instance| declaration. One you have declared your archetype, you can describe how it should be model checked using this syntax:
+
+\begin{lstlisting}]language=pcal]
+process (YourInstanceName = 1) == instance YourArchetypeName(ref param1, param2, ...)
+\end{lstlisting}
+
+This will define a process called \lstinline|YourInstanceName| with \lstinline|self=1|, delegating all information about the process body to the archetype \lstinline|YourArchetypeName|. Just like with PlusCal processes, you can define a set of concurrently executing processes by specifying \lstinline[language=pcal]|YourInstanceName \in someset|. You can also define multiple separate processes as instances of the same archetype if needed.
+
+The parameters \lstinline|ref param1| and \lstinline|param2| show the two possible syntaxes for parameter passing. In either case, \lstinline|param1| and \lstinline|param2| are required to be already-declared PlusCal global variables.
+
+This instance declatation deliberately matches the syntax example for archetype declaration declaring \lstinline|YourArchetypeName|, which shows that in order to pass an archetype parameter by reference (see section \ref{passbyref}) both the parameter declaration and the value passed in must be declared \lstinline|ref|. This is so that it is clear at both instantiation and declaration that that parameter can be used to mutate the environment.
+
+For a more fleshed out example, consider:
+
+\begin{lstlisting}[language=pcal]
+CONSTANTS COORDINATORS, BACKUPS \* this is declared in the TLA+ code, but is written here for brevity
+
+variables connection = <<>>,
+           backupConnection = <<>>;
+
+process (MainCoordinator \in COORDINATORS) == instance Coordinator(connection);
+process (BackupCoordinator \in BACKUPS) == instance Coordinator(backupConnection);
+\end{lstlisting}
+
+In the definition above, the connection variable is global in PlusCal. However, when PGo compiles an specification like the one above, only source code for archetypes is generated. Archetype parameters represent implementation-specific details that need to be filled in by the developer (oftentimes, the PGo runtime will provide most of the logic required in these implementation-specific components).
+
+\subsubsection{Instantiating the Go implementation}
+
+TODO I don't think we have this down yet
```

Comment:
@mrordinaire what should we do about this part?
Since we don't really have codegen set up yet, probably just leave it as is?
Just confirming my intuition.

Reply:
agreed. this will come after Go codegen.